extensions/amp-story-shopping/0.1/amp-story-shopping-tag.js

```diff
@@ -38,9 +38,7 @@ const renderShoppingTagTemplate = (tagData) => (
         }
       ></span>
       <span class="amp-story-shopping-tag-pill-text">
-        {tagData['product-tag-text']
-          ? tagData['product-tag-text']
-          : tagData['product-price']}
+        {tagData['product-tag-text'] ?? tagData['product-price']}
```

Comment:
What is the benefit to using `??` rather than `||` in this case?

Reply:
`??` will keep the first value if it's an empty string (it just grabs the second value if the first is `null` or `undefined`). `||` keeps the second value if the first value is falsey. If you want to avoid empty strings, `||` makes more sense, but if you want to allow them (and not show the price), `??` is the one to use. Probably `||` is a better fit for this since it will ignore empty strings and show the tag.

Reply:
?? uses the right hand expression if the left hand evaluates to null or undefined.

|| uses the right hand expression if the left hand evaluates to a [falsy](https://developer.mozilla.org/en-US/docs/Glossary/Falsy) value.

They would both work, but || performs additional checks for these falsy values (so we cannot have falsy values as product names).

In our case, we are concerned with the null / undefined case. 
So ?? makes more sense to use in this particular case.

Reply:
alright, good point, I've updated it to use ||

Reply:
What if the user enters an empty string? wouldn't it render an a tag without text? We don't want that.

Reply:
Empty strings when using `||` are falsey so the code would use the second value. Pubs could add the string `" "` (one space) to render it empty, or use other ways to cheat this, so optionally we could use [.trim()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/Trim) but up to you 😁. Probably there will always be ways to cheat this but I don't see why they'd not want to show the tag with a name or price.

Reply:
I'm for keeping it as `||`, I think adding .trim(), would add another null check (as we don't know if the string is undefined). This would just make the code a bit harder to read for the unlikely edge case that Matias mentions above.

Reply:
However, @processprocess the dollar sign shows up even when the product name is displayed. It names sense to show up on the product price, but not the name, should we add another ticket to take care of this?

Reply:
Thanks for finding that :) We can't merge it as is, that would introduce a bug.
We need to refactor that in this PR.

A couple ideas / approaches: 
- Add the dollar sign in the template rather than as a pseudo element.
- Append a class if it's text and not the price.

Reply:
I think it would make more sense to make $ an internationalized string so it can be localized to different currencies (¥,£, etc.) right?

Reply:
It's impossible to tell from the language what currency people use (spanish is used in 20+ languages for example), so it could be either #1: part of the string they pass (eg: `"$54", "54USD"`) or #2: a separate config (eg: `"currency": "USD"`) but you don't know how to format it. Probably #1 makes more sense?

Reply:
#2 make more sense to me - I feel like if we are adding in more things to the string, that is adding inconvenience for the user (they would have to perform that operation for every string - I would rather focus on ease of use and add currency as a config option.).

I've seen some other libraries use "prefix" and "suffix"
prefix : "$"

Reply:
We are sticking with usd for V1. This was noted in the design [mocks here](https://www.figma.com/file/QfezXvJWxaixCjkk7RRGvM?node-id=402:6396#113387626).

Questions like this need to be discussed with PM. We cannot make decisions like this in PRs.

Reply:
Ok, then this PR is good to go then. we can bring this up in the next shopping sync.

Reply:
Filed #37126 for tracking.